repair/row_level.cc

```diff
@@ -2925,7 +2925,7 @@ class row_level_repair {
             throw;
           }
         }
-        master.flush_rows_in_working_row_buf(_small_table_optimization ? get_erm() : nullptr, _small_table_optimization);
+        master.flush_rows_in_working_row_buf(_small_table_optimization ? get_erm().copy().get() : nullptr, _small_table_optimization);
```

Comment:
This introduces a cross-shard trip here. Can this be avoided?

Reply:
In 98323be296fdc96c6f63e15fa196c6de0ae349af we started waiting until migrations are done and executing tablet repair with the first safe topology version. This is done on a shard that initiates the repair and erm is kept on this shard for the whole repair.

In this PR we want to release erm when possible and the responsibility for it is moved to shard repair task (it needs to have access to erm), which updates erm if it is compatible with the previous erm on its tablets. A shard task has to start with the initial erm and can't get its own local version of erm as it might have already changed on this shard.

So we have to keep a foreign pointer to erm. We cannot pass an erm reference as shard task might update it during `flush_rows_in_working_row_buf` execution.

Reply:
I don't mind using `foreign_ptr`, I would like to avoid `foreign_ptr::copy()`, if possible. This method will result in a cross-shard trip. If not possible, then fine.

Reply:
I don't think it is. But, after the first erm update, this shard would be the owner of the pointer